runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -161,6 +161,22 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
                 return TransactionalContext.getCurrentContext()
                         .access(this, accessMethod, conflictObject);
             } catch (Exception e) {
+                if (e instanceof TrimmedException) {
```

Comment:
This doesn't seem needed - the TXcontext will reset the object, and under a lock since it is done inside VersionLockedObject::access’s call to updateFunction. 

Perhaps though, rather than aborting the TXContext itself needs to retry at least one to pick up the checkpoint.